src/test/ui/span/unused-warning-point-at-signature.stderr

```diff
@@ -0,0 +1,36 @@
+warning: enum is never used: `Enum`
+  --> $DIR/unused-warning-point-at-signature.rs:15:1
+   |
+15 | enum Enum {
+   | ^^^^^^^^^
+   |
+note: lint level defined here
+  --> $DIR/unused-warning-point-at-signature.rs:13:9
+   |
+13 | #![warn(unused)]
+   |         ^^^^^^
+   = note: #[warn(dead_code)] implied by #[warn(unused)]
+
+warning: struct is never used: `Struct`
+  --> $DIR/unused-warning-point-at-signature.rs:22:1
+   |
+22 | struct Struct {
+   | ^^^^^^^^^^^^^
+
+warning: function is never used: `func`
+  --> $DIR/unused-warning-point-at-signature.rs:29:1
+   |
+29 | fn func() -> usize {
+   | ^^^^^^^^^^^^^^^^^^
+
+warning: function is never used: `func_complete_span`
+  --> $DIR/unused-warning-point-at-signature.rs:33:1
+   |
+33 | / fn
+34 | | func_complete_span()
+35 | | -> usize
+36 | | {
+37 | |     3
+38 | | }
```

Comment:
Why is this span including the body? Just because it's multiline?

Reply:
@oli-obk it was originally proposed that given that in cases where a multiline span was already being shown, we might as well point at the entire body. This decision can be revised if needed.

Reply:
For RLS this is a major annoyance.

Reply:
@oli-obk For RLS indeed it'd be very handy to only point at the symbol name, but I think from the CLI ergonomics point of view, displaying multiline diagnostics might still be superior? (gives more context, CLIs are often run out-of-the-editor etc.)

Reply:
It would still be multiline, it just wouldn't show the function body. I don't see much use in showing function bodies at all for such diagnostics

Reply:
Ah, multiline function signature (without the body), got it. Makes sense, thanks :+1:

Reply:
@oli-obk, @Xanewok Only one line of code would need to chance: https://github.com/rust-lang/rust/blob/fd8099f0ecc2fd3980469ec45cf834ae9a941ca3/src/libsyntax/codemap.rs#L446 (remove the check for no '\n').

Shall we file ticket for it?

Reply:
If that would work, I guess we could. However, aren't there other `def_span` users who still rely on the full multiline when a signature is also spanning across multiple lines? I'm not very well versed in rustc itself, sorry.

Reply:
@Xanewok there're are only two users of `def_span` at the moment, including this one, and neither _rely_ on it showing the full span when multiline.

Reply:
@estebank ah, my bad. Must've looked up wrong `def_span`, sorry. The one from the codemap seems to only be used for lints (if I'm not mistaken?) so I think it'd be great to go ahead and limit the multiline spans only to signatures, without bodies, in lints.

Reply:
@estebank @oli-obk with the [change](https://github.com/rust-lang/rust/pull/44847#discussion_r142738403) now also `ui\span\multiline-span-E0072.rs` fails as follows:
after change:
```
error[E0072]: recursive type `ListNode` has infinite size
  --> $DIR/multiline-span-E0072.rs:12:1
   |
12 | / struct
13 | | ListNode
   | |________^ recursive type has infinite size
...
16 |       tail: Option<ListNode>,
   |       ---------------------- recursive without indirection
   |
   = help: insert indirection (e.g., a `Box`, `Rc`, or `&`) at some point to make `ListNode` representable
```
expected:
```
error[E0072]: recursive type `ListNode` has infinite size
  --> $DIR/multiline-span-E0072.rs:12:1
   |
12 | / struct
13 | | ListNode
14 | | {
15 | |     head: u8,
16 | |     tail: Option<ListNode>,
   | |     ---------------------- recursive without indirection
17 | | }
   | |_^ recursive type has infinite size
   |
   = help: insert indirection (e.g., a `Box`, `Rc`, or `&`) at some point to make `ListNode` representable
```

Is this desired? On one hand it's handy to highlight the faulty (recursive here) bit within the whole span, on the other highlighting *entire* definition may be an overkill, especially for RLS (iirc right now RLS can't even highlight primary/secondary spans properly per a single diagnostic, only the primary one).

Reply:
@Xanewok I also prefer the current output, but understand that we're bound by the limitations of VSCode/RLS here.

@nikomatsakis would you have a comment? I'm inclined to move ahead with the change.